### ISSUE TITLE
Use of deprecated CookieSyncManager

### ISSUE THREAD

**Issue by frankFic** [TwidereProject/Twidere-Android#974]
Hi there,

I noticed that you are still using the deprecated CookieSyncManager [here](https://github.com/TwidereProject/Twidere-Android/blob/3948954973a09de5fc7250741a19b0d24bb0cc38/twidere/src/main/kotlin/org/mariotaku/twidere/activity/SignInActivity.kt#L159). The class has been deprecated since API 21 according to the [doc](https://developer.android.com/reference/android/webkit/CookieSyncManager.html).
I also noticed that removeAllCookiesSupport(), also deprecated since API 21, is used.

I also noticed that a wrapper method that handles the deprecation problem for these APIs has been implemented in [CookieManagerExtension.kt](https://github.com/TwidereProject/Twidere-Android/blob/d19671085c5c195496f42749d2f195d921a1fc5c/twidere/src/main/kotlin/org/mariotaku/ktextension/CookieManagerExtension.kt). I am just wondering why not just use this wrapper method?

Thx.

**Comment by mariotaku** [comment]
@frankFic Hi, calling `CookieSyncMananger.createInstance()` is for compatibility. `CookieManager.getInstance()` may throw exceptions without `CookieSyncMananger.createInstance()` called on some versions.

**Comment by frankFic** [comment]
@mariotaku Thanks for your quick reply! I know that on old versions without `CookieSyncMananger.createInstance()`, `CookieManager.getInstance()` may throw exception. But this problem has been fixed since API 21. I also found a related issue on Android issue tracker: [https://issuetracker.google.com/issues/37014570](https://issuetracker.google.com/issues/37014570).  The Android developer also indicated that the problem of CookieManager has been fixed and changed the [document](https://developer.android.com/reference/android/webkit/CookieManager.html#getInstance()). So I think it may not be necessary to use this deprecated class on higher API levels.
